lib/dbPool.js

```diff
@@ -190,6 +193,21 @@ class DBPool {
     }
   }
 
+  /**
+   * @description
+   * Configuration option to return numeric types as a Number instead of a String.
+   * When set to true numeric types for `DBPool.runSql` and `DBPool.prepareExecute` 
+   * will be returned as a Number instead of a String.
+   * @param {boolean} flag - option to return numeric types as a Number instead of a String
+   * @returns {boolean} - resolves to true/false indicating the state of the flag
+   */
+  enableNumericTypeConversion(flag) {
```

Comment:
This method name conflicts with the `enableNumericTypeConversion` property. Therefore one or the other needs to be renamed.  As it is now the following error will occur:

```js
 const pool = new DBPool();
 pool.enableNumericTypeConversion(true);
```

```
 TypeError: pool.enableNumericTypeConversion is not a function

```

I would recommend renaming the property from `enableNumericTypeConverion` to `numericTypeConversion` and leaving the method name as is (enableNumericTypeConversinon).

Reply:
```diff
diff --git a/lib/dbPool.js b/lib/dbPool.js
index 9fb0c4f..c1b5691 100644
--- a/lib/dbPool.js
+++ b/lib/dbPool.js
@@ -9,7 +9,8 @@ class DBPool {
    * Manages a list of DBPoolConnection instances.
    * Constructor to instantiate a new instance of a DBPool class given the `database` and `config`
    * @param {object} database - Object includes the `url`, `username`, and `password`.
-   * @param {object} config - Object includes the `incrementSize`, `debug` and `enableNumericTypeConversion`.
+   * @param {object} config - Object includes the `incrementSize`, `debug` and
+   * `numericTypeConversion`.
    * @constructor
    */
   constructor(database = {
@@ -17,12 +18,12 @@ class DBPool {
   }, config = {
     incrementSize: 8,
     debug: false,
-    enableNumericTypeConversion: false,
+    numericTypeConversion: false,
   }) {
     const {
       incrementSize,
       debug,
-      enableNumericTypeConversion,
+      numericTypeConversion,
     } = config;
 
     this.connections = [];
@@ -33,7 +34,7 @@ class DBPool {
     };
     this.incrementSize = incrementSize || 8;
     this.debug = debug || false;
-    this.enableNumericTypeConversion = enableNumericTypeConversion || false;
+    this.numericTypeConversion = numericTypeConversion || false;
 
     for (let i = 0; i < this.incrementSize; i += 1) {
       this.createConnection(i);
@@ -196,16 +197,16 @@ class DBPool {
   /**
    * @description
    * Configuration option to return numeric types as a Number instead of a String.
-   * When set to true numeric types for `DBPool.runSql` and `DBPool.prepareExecute` 
+   * When set to true numeric types for `DBPool.runSql` and `DBPool.prepareExecute`
    * will be returned as a Number instead of a String.
    * @param {boolean} flag - option to return numeric types as a Number instead of a String
    * @returns {boolean} - resolves to true/false indicating the state of the flag
    */
   enableNumericTypeConversion(flag) {
-    if(typeof flag !== 'undefined') {
-      this.enableNumericTypeConversion = flag && true;
+    if (typeof flag === 'boolean') {
+      this.numericTypeConversion = flag;
     }
-    return this.enableNumericTypeConversion;
+    return this.numericTypeConversion;
   }
 
   /**
@@ -220,7 +221,7 @@ class DBPool {
     let resultSet = [];
     let reason;
 
-    if(this.enableNumericTypeConversion) {
+    if (this.numericTypeConversion) {
       statement.enableNumericTypeConversion(true);
     }
 
@@ -268,7 +269,7 @@ class DBPool {
     let outParams = null;
     let reason;
 
-    if(this.enableNumericTypeConversion) {
+    if (this.numericTypeConversion) {
       statement.enableNumericTypeConversion(true);
     }

```

Reply:
Hello @patrickhrastnik just pinging you to see if you can enable [allow changes to a pull request branch created from a fork](https://docs.github.com/en/free-pro-team@latest/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork)?

That way I can make some suggested changes and add test cases to this PR.

Reply:
Thanks for the changes @abmusse 
I can't find this option in this PR, but I'll have a look into it so you can push your changes

Reply:
@abmusse I added you as a member of HolterDev/nodejs-idb-pconnector
hope that helps

Reply:
OK Thanks! 

This should work I will try to push my changes there.